app/models/driver.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'trip'
+class Driver < ApplicationRecord
+  has_many :trips
+
+  def total_earnings
+    trips = self.trips
+    sum = 0
+    trips.each do |trip|
+      if trip.cost
+        app_cut = trip.cost * (0.15)
+        sum += trip.cost - app_cut
+      end
+    end
+    return sum
+  end
+
+  def average_rating
+    total = 0.0
+    trips = self.trips
+    trip_count = trips.length
+    trips.each do |trip|
+      total += trip.rating
+    end
+    return "#{(total / trip_count).round(2)}"
```

Comment:
What happens if there are no trips associated with a driver? Then `total` would be 0.0 and `trip_count` is 0. Ruby evaluates `0.0 / 0` as `NaN` or "Not A Number." When it gets to the driver/show.html.erb view, `NaN.to_s` evaluates to just "NaN" so it shows up on the driver page as "NaN" on the page. It'll show up like this:
"Minnie Dach

Average rating is NaN out of 5."